src/Strategy/RPM.hs

```diff
@@ -0,0 +1,137 @@
+module Strategy.RPM
+  ( discover,
+    getSpecDeps,
+    getTypeFromLine,
+    RPMDependency (..),
+    RequiresType (..),
+    Dependencies (..),
+  )
+where
+
+import Control.Effect.Error
+import qualified Data.Map.Strict as M
+import Data.Maybe (mapMaybe)
+import qualified Data.Text as T
+import DepTypes
+import Discovery.Walk
+import Effect.Grapher
+import Effect.ReadFS
+import Graphing (Graphing)
+import Prologue
+import Types
+
+newtype SpecFileLabel
+  = RequiresType DepEnvironment
+  deriving (Eq, Ord, Show, Generic)
+
+data RPMDependency
+  = RPMDependency
+      { rpmDepName :: Text,
+        rpmConstraint :: Maybe VerConstraint
+      }
+  deriving (Eq, Ord, Show, Generic)
+
+data RequiresType
+  = BuildRequires RPMDependency
+  | RuntimeRequires RPMDependency
+  deriving (Eq, Ord, Show, Generic)
+
+data Dependencies
+  = Dependencies
+      { depBuildRequires :: [RPMDependency],
+        depRuntimeRequires :: [RPMDependency]
+      }
+  deriving (Eq, Ord, Show, Generic)
+
+discover :: HasDiscover sig m => Path Abs Dir -> m ()
+discover = walk $ \dir _ files ->
+  let specs = find (\f -> ".spec" `isSuffixOf` fileName f) files
+      analzyeFile specFile = runSimpleStrategy "rpm-spec" RPMGroup $ fmap (mkProjectClosure dir) (analyze specFile)
+   in traverse_ analzyeFile specs >> pure WalkSkipAll
+
+analyze :: (Has ReadFS sig m, Has (Error ReadFSErr) sig m) => Path Rel File -> m (Graphing Dependency)
+analyze specFile = do
+  specFileText <- readContentsText specFile
+  pure . buildGraph $ getSpecDeps specFileText
+
+mkProjectClosure :: Path Rel Dir -> Graphing Dependency -> ProjectClosureBody
+mkProjectClosure dir graph =
+  ProjectClosureBody
+    { bodyModuleDir = dir,
+      bodyDependencies = dependencies,
+      bodyLicenses = []
+    }
+  where
+    dependencies =
+      ProjectDependencies
+        { dependenciesGraph = graph,
+          dependenciesOptimal = NotOptimal,
+          dependenciesComplete = NotComplete
+        }
+
+toDependency :: RPMDependency -> Set a -> Dependency
+toDependency pkg =
+  foldr
+    (const id)
+    Dependency
+      { dependencyType = RPMType,
+        dependencyName = rpmDepName pkg,
+        dependencyVersion = rpmConstraint pkg,
+        dependencyLocations = [],
+        dependencyEnvironments = [],
+        dependencyTags = M.empty
+      }
+
+buildGraph :: Dependencies -> Graphing Dependency
+buildGraph Dependencies {..} = run . withLabeling toDependency $ traverse direct depBuildRequires
```

Comment:
@cnr Is there a better way to run the grapher if you aren't using labels?

Reply:
You can:
1. `evalGrapher`, which will give you `m (Graphing RPMDependency)`
2. Use `gmap :: (Ord ty, Ord ty') => (ty -> ty') -> Graphing ty -> Graphing ty'` to change the type of elements in the `Graphing`

Reply:
Oh another option here for building the graph: because you just have a list of direct dependencies, `Graphing.fromList :: Ord a => [a] -> Graphing a` is probably what you want

Reply:
I originally tried `evalGrapher`, but I missed `gmap`, so I didn't get it to work.  `fromList` is definitely the move here, although I don't know if that will work if/when we include `Requires` dependencies.